src/libcollections/linked_list.rs

```diff
@@ -898,9 +898,9 @@ impl<T> IntoIterator for LinkedList<T> {
 #[stable(feature = "rust1", since = "1.0.0")]
 impl<'a, T> IntoIterator for &'a LinkedList<T> {
     type Item = &'a T;
-    type IntoIter = Iter<'a, T>;
+    type IntoIter = Iter<'a, Self::IntoIter>;
```

Comment:
This should have still been T, and is causing infinite recursion in type checking.